lib/guest_configuration_base.pm

```diff
@@ -0,0 +1,1408 @@
+                                                      #[guest_version_major]:15 or others, not virt-install argument
+                                                      #[guest_version_minor]:3 or others, not virt-intall argument
+                                                      #The following five parameters are detailed guest os information,not virt-install arguments
+    'guest_os_name'                         => '',
+    'guest_os_word_length'                  => '',
+    'guest_version'                         => '',
+    'guest_version_major'                   => '',
+    'guest_version_minor'                   => '',
+    'guest_build'                           => '',    #Build number if developing release or 'GM'
+    'guest_log_folder'                      => '',    #Log folder for individual guest [common_log_folder]/[guest_name]
+    'guest_installation_result'             => '',    #PASSED,FAILED,TIMEOUT,UNKNOWN or others
+    'guest_installation_session' => '', #Guest installation process started by "screen -t screen_title [virt_install_command_line]. It is in the form of 3401.pts-1.vh017
+    'guest_installation_session_command' => '', #If there is no [guest_installation_session] or [guest_installation_session] is terminated, start or re-connect to
+        #guest installation screen using [guest_installation_session_command] = screen -t [guest_name] virsh console --force [guest_name]
+    'guest_installation_attached' => '',    #This indicates whether guest installation screen is already connected or attached(true or false)
+    'start_run'                   => '',    #Guest creation start time
+    'stop_run'                    => ''     #Guest creation finish time
+);
+
+our $AUTOLOAD;
+our $common_log_folder           = '/tmp/guest_installations';
+our $common_environment_prepared = 'false';
+
+#Any subroutine calls this subroutine announces its identity and it is being executed
+sub reveal_myself {
+    my $self = shift;
+
+    my $_my_identity = (caller(1))[3];
+    diag("Test execution inside $_my_identity.");
+    return $self;
+}
+
+#Create guest instance by assigning values to its parameters but do no install it
+sub create {
+    my $self = shift;
+
+    $self->reveal_myself;
+    $self->initialize_guest_params;
+    $self->config_guest_params(@_);
+    $self->print_guest_params;
+    return $self;
+}
+
+#Initialize all guest parameters to avoid uninitialized parameters
+sub initialize_guest_params {
+    my $self = shift;
+
+    $self->reveal_myself;
+    $self->{$_} //= '' foreach (keys %guest_params);
+    $self->{host_ipaddr} = get_required_var('SUT_IP');
+    $self->{host_name}   = script_output("hostname");
+    $self->{start_run}   = time();
+    return $self;
+}
+
+#Assign real values to guest instance parameters.Reset [guest_name] to guest name used in [guest_metadata] if they are different.
+sub config_guest_params {
```

Comment:
It will be helpful to give some call example for this function , so that in future other users that make new types of guest installation can know the usage directly.